src/asl.ts

```diff
@@ -1516,6 +1535,72 @@ export class ASL {
           return this.eval(values.expr);
         }
         throw new SynthError(ErrorCodes.Unsupported_Use_of_Promises);
+      } else if (
+        isPropAccessExpr(expr.expr) &&
+        isIdentifier(expr.expr.expr) &&
+        expr.expr.expr.name === "JSON"
+      ) {
+        if (expr.expr.name === "stringify") {
+          const heap = this.newHeapVariable();
+
+          const objParamExpr = expr.args[0].expr;
+          if (!objParamExpr || isUndefinedLiteralExpr(objParamExpr)) {
+            // return an undefined variable
+            return {
+              jsonPath: heap,
+            };
+          }
+
+          return this.evalExpr(
+            objParamExpr,
+            (_, { normalizeOutputToJsonPath }) => {
+              const objectPath = normalizeOutputToJsonPath().jsonPath;
+
+              return {
+                Type: "Pass",
+                Parameters: {
+                  // intrinsic functions cannot be used in InputPath and some other json path locations.
+                  // We compute the value and place it on the heap.
+                  "string.$": `States.JsonToString(${objectPath})`,
+                },
+                ResultPath: heap,
+                output: {
+                  jsonPath: `${heap}.string`,
+                },
+              };
+            }
+          );
+        } else if (expr.expr.name === "parse") {
+          const heap = this.newHeapVariable();
+
+          const objParamExpr = expr.args[0].expr;
+          if (!objParamExpr || isUndefinedLiteralExpr(objParamExpr)) {
+            // return an undefined variable
+            return {
+              jsonPath: heap,
+            };
+          }
+
+          return this.evalExpr(
+            objParamExpr,
+            (_, { normalizeOutputToJsonPath }) => {
+              const objectPath = normalizeOutputToJsonPath().jsonPath;
+
+              return {
+                Type: "Pass",
+                Parameters: {
+                  // intrinsic functions cannot be used in InputPath and some other json path locations.
+                  // We compute the value and place it on the heap.
+                  "string.$": `States.StringToJson(${objectPath})`,
+                },
+                ResultPath: heap,
+                output: {
+                  jsonPath: `${heap}.string`,
+                },
+              };
+            }
+          );
+        }
```

Comment:
Looks almost exactly the same as JSON.stringify

Reply:
hmm, they did end up similar. will merge.